src/api/base.ts

```diff
@@ -0,0 +1,59 @@
+import ky from 'ky';
+
+import { ERROR_MESSAGE } from '../utils/constants';
+import { ApiError } from '../types';
+
+// const apiUrl = process.env.NEXT_PUBLIC_API_URL;
+const apiUrl = 'https://sportujspolu-api.onrender.com/api/v1/';
+
+export const api = ky.create({
+  prefixUrl: apiUrl,
+  headers: {},
+});
+
+export const privateApi = (token: string) =>
+  ky.create({
+    prefixUrl: apiUrl,
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+export const apiPost = async <T>(
+  endpoint: string,
+  body: unknown,
+  token?: string,
+  onError?: (error: ApiError) => void,
+): Promise<T> => {
+  const apiInstance = token ? privateApi(token) : api;
+
+  try {
+    return await apiInstance.post(endpoint, { json: body }).json();
+  } catch (error: unknown) {
+    const apiError: ApiError = {
+      status: (error as { response?: { status: number } }).response?.status,
+      message: ERROR_MESSAGE.GENERIC_ERROR,
+    };
+    onError?.(apiError);
+    throw apiError;
```

Comment:
Could this be better? Casting and any is quite the same. This is just less code

```suggestion
 } catch (error: any) {
    onError?.({
      status: error.response?.status,
      message: ERROR_MESSAGE.GENERIC_ERROR,
    });
  }
  
      return <- well we will have to return something. What should it be? do we want to have easy `{}` empty object or rather `null` ? Not sure. I went for `null` in my project, but I am not sure it is the silver bullet
```